pandas/tests/frame/test_constructors.py

```diff
@@ -2548,8 +2555,20 @@ def check_views(c_only: bool = False):
             check_views()
 
         # TODO: most of the rest of this test belongs in indexing tests
-        df.iloc[0, 0] = 0
-        df.iloc[0, 1] = 0
+        if (
+            is_float_dtype(df.dtypes.iloc[0])
+            or is_integer_dtype(df.dtypes.iloc[0])
+            or is_complex_dtype(df.dtypes.iloc[0])
+            or is_object_dtype(df.dtypes.iloc[0])
+            or is_datetime64_dtype(df.dtypes.iloc[0])  # TODO this one should warn
+            or is_timedelta64_dtype(df.dtypes.iloc[0])  # TODO this one should warn
```

Comment:
are these TODOs things you want to do for this PR or follow-ups?

Reply:
this one addresses them: https://github.com/pandas-dev/pandas/pull/53405